File: app/src/main/java/com/edison/librolink/data/retrofit/RetrofitInstance.kt
package com.edison.librolink.data.retrofit

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitInstance {
    val api: GoogleBooksApi by lazy {
        Retrofit.Builder()
            .baseUrl("https://www.googleapis.com/books/v1/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(GoogleBooksApi::class.java)
    }
}

File: lib/data/retrofit/retrofit_instance.dart
import 'package:dio/dio.dart';
import 'package:libro_link/data/retrofit/google_books_api.dart';
class ApiClient {
  static final Dio _dio = Dio(BaseOptions(
    baseUrl: 'https://www.googleapis.com/books/v1/',
    connectTimeout: const Duration(seconds: 10),
    receiveTimeout: const Duration(seconds: 10),
  ));

  static final GoogleBooksApi api = GoogleBooksApi(_dio);
}
